lib/services/database/upvote_repository_service.dart

```diff
@@ -86,13 +87,13 @@ class UpvoteRepositoryService<ParentIdFirestore extends IdFirestore> {
         ? await transaction.get(voteStateCollection)
         : await voteStateCollection.get();
 
-    if (!voteState.exists) {
-      return VoteState.none;
-    } else {
+    VoteState getActualState() {
```

Comment:
Why is this a local function instead of a final variable ?

Reply:
I think it is because otherwise `voteState.data()` might be null and therefore causing a null pointer exception on the `!` call right ?

If this is the reason why I think some documentation should be added to better understand this.

Reply:
Or maybe we could keep the original guard clause on this:

```dart
if (!voteState.exists) {
    return VoteState.none;
}
```

and simply remove the `else` clause.

I think that way the code should be more readable, and I don't think DCM would complain.

Reply:
I tried a new implementation with [9ebc786](https://github.com/ProximaEPFL/proxima/pull/262/commits/9ebc786a288ead702f4192a3fab954ea1326d700). What do you think about it?

Reply:
Yes, much better ty :+1: